Add reference docs for The Pop-Up API, and my own author details (#4069)
GoogleChrome/developer.chrome.com: site/en/docs/web-platform/popup-api/index.md

* `manual`: Manual pop-ups cannot be light dismissed (they can only be dismissed by an explicit trigger element or by JavaScript), and they don't automatically dismiss previously-shown pop-ups.

You can show and dismiss pop-ups in two different ways:

* Declaratively, using HTML attributes. Add the [`popuptoggletarget`](/docs/web-platform/popup-api/popuptoggletarget-attribute), [`popupshowtarget`](/docs/web-platform/popup-api/popupshowtarget-attribute), or [`popuphidetarget`](/docs/web-platform/popup-api/popuphidetarget-attribute) to an `<input>` or `<button>` element to turn it into a pop-up trigger element. In each case, the attribute value must be the `id` of the pop-up, to associate the two.
* Programmatically, using JavaScript. The [`showPopUp()`](/docs/web-platform/popup-api/showpopup-method) and [`hidePopUp()`](/docs/web-platform/popup-api/hidepopup-method) methods are used to show and dismiss pop-ups, respectively, and the [`popupshow`](/docs/web-platform/popup-api/popupshow-event) and [`popuphide`](/docs/web-platform/popup-api/popuphide-event) events can be used to react to pop-ups being shown or hidden.

> The Pop-Up API also comes with a couple of handy CSS features, a [`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element) pseudo-element to style the document behind the pop-up when it is shown (e.g. you might want to blur or fade the page while showing pop-ups), and an [`:open`](/docs/web-platform/popup-api/open-pseudo-class) pseudo-class to style the pop-up only when it is open.

Reviewer: ```suggestion
The Pop-Up API also comes with a couple of handy CSS features, a [`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element) pseudo-element to style the document behind the pop-up when it is shown (for example, you might want to blur or fade the page while showing pop-ups), and an [`:open`](/docs/web-platform/popup-api/open-pseudo-class) pseudo-class to style the pop-up only when it is open.
```

## HTML attributes

[`defaultopen`](/docs/web-platform/popup-api/defaultopen-attribute)
: Causes a pop-up to automatically open on page load.

[`popup`](/docs/web-platform/popup-api/popup-attribute)
: Turns an element into a pop-up.

[`popuptoggletarget`](/docs/web-platform/popup-api/popuptoggletarget-attribute)
: Creates a trigger element that toggles an associated pop-up between shown and hidden states.

[`popupshowtarget`](/docs/web-platform/popup-api/popupshowtarget-attribute)
: Creates a trigger element that shows an associated hidden pop-up.

[`popuphidetarget`](/docs/web-platform/popup-api/popuphidetarget-attribute)
: Creates a trigger element that dismisses an associated shown pop-up.

## CSS features

[`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element)

> : Pseudo-element that sits behind an open pop-up in the stacking order, but in front of the rest of the document, and spans the entire width and height of the viewport. Allows the rest of the content to be styled while the pop-up is open — for example you might want to blur or darken it.

Reviewer: ```suggestion
: Pseudo-element that sits behind an open pop-up in the stacking order, but in front of the rest of the document, and spans the entire width and height of the viewport. Allows the rest of the content to be styled while the pop-up is open—for example you might want to blur or darken it.
```

{% Aside %}
This section shows basic usage of the API; for more substantial code including numerous complete examples, check out [Pop-ups: They're making a resurgence!](/blog/pop-ups-theyre-making-a-resurgence/) by Jhey Tompkins.
{% endAside %}

Declaratively create different types of pop-up. The first two are equivalent:

```html
<p id="my-popup" popup>Hello!</p>
<p id="my-popup" popup="auto">Hello!</p>
<p id="my-popup" popup="hint">Hello!</p>
<p id="my-popup" popup="manual">Hello!</p>
```

Specify that a pop-up should open automatically on page load with [`defaultopen`](/docs/web-platform/popup-api/defaultopen-attribute):

```html
<p id="my-popup" popup="auto" defaultopen>Hello!</p>
```

{% Aside 'caution' %}
[`defaultopen`](/docs/web-platform/popup-api/defaultopen-attribute) does not work with pop-ups of type `hint`.
{% endAside %}

Declaratively create buttons to show and hide the pop-up:

```html
<button popupshowtarget="my-popup">Show pop-up</button>
<button popuphidetarget="my-popup">Dismiss pop-up</button>
```

Declaratively create a single button to toggle the pop-up between shown and hidden states:

```html
<button popuptoggletarget="my-popup">Show pop-up</button>
```

> Use the [`:open`](/docs/web-platform/popup-api/open-pseudo-class) pseudo-class to style the pop-up when it has been shown. In this case we are transitioning the pop-up in from the bottom of the viewport, rather than just having it appear in the center:

Reviewer: ```suggestion
Use the [`:open`](/docs/web-platform/popup-api/open-pseudo-class) pseudo-class to style the pop-up when it has been shown. The following example transitions the pop-up in from the bottom of the viewport, rather than have it appear in the center:
```

```css
p[popup] {
  transform: translateY(60vh);
}

p[popup]:open {
  transition: transform 0.6s;
  transform: translateY(0vh);
}
```

> Use the [`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element) pseudo-element to style the document behind the pop-up while it is being shown — here we are darkening the rest of the content:

Reviewer: ```suggestion
Use the [`::backdrop`](/docs/web-platform/popup-api/backdrop-pseudo-element) pseudo-element to style the document behind the pop-up while it is being shown. The following example darkens the rest of the content:
```

```css
p[popup]::backdrop {
  background-color: rgba(0,0,0,0.5);
}
```

Use JavaScript to programmatically show and hide pop-ups:

```js
  popup.showPopUp();
  popup.hidePopUp();
```

> Use event listeners to respond to pop-ups being shown and hidden. Here we are switching the `textContent` of the pop-up trigger button so that it makes sense when the pop-up is in its shown and hidden states:

Reviewer: ```suggestion
Use event listeners to respond to pop-ups being shown and hidden. The following example switches the `textContent` of the pop-up trigger button so that it makes sense when the pop-up is in its shown and hidden states:
```
